File: output/TCOFFI.php
<?php
$exports['runTCOFFI'] = function($limit) {
    $go = function($n, $acc) use (&$go) {
        if ($n <= 0) return $acc;
        return $go($n - 1, $acc + $n);
    };
    return $go((int)$limit, 0);
};
return $exports;

File: scratch/gopurs-adt-reuse-validation-20260916/before-fficc/src/Test/TCOFFI.cc
#include "purescript.h"

namespace {
    int deepTailRec(int n, int acc) {
        if (n == 0) return acc;
        return deepTailRec(n - 1, acc + n % 3);
    }
}

FOREIGN_BEGIN(Test_TCOFFI)
exports["runTCOFFI"] = [](const boxed& in) -> boxed {
    const int result = deepTailRec(unbox<int>(in), 0);
    return result;
};
FOREIGN_END
